src/foundry/client/data/documents/cards.d.mts

```diff
@@ -288,9 +289,17 @@ declare global {
       context: Record<string, unknown>,
     ): Promise<ChatMessage.ConfiguredInstance | undefined>;
 
-    protected override _onUpdate(
+    protected override _preCreate(
+      // can't use UpdateData because it creates circular reference
       data: fields.SchemaField.InnerAssignmentType<BaseCards.Schema>,
       options: DocumentModificationOptions,
+      user: BaseUser,
+    ): Promise<boolean | void>;
+
+    protected override _onUpdate(
+      // can't use UpdateData because it creates circular reference
+      changed: fields.SchemaField.InnerAssignmentType<BaseCards.Schema>,
```

Comment:
Is this `_onUpdate` note filed as a note on an issue anywhere? This question can probably be answered by @JPMeehan too.

I want to make sure I get to all wrinkles like this eventually because they're all suspicious

Reply:
I think this is just a side effect of our documents with `ForeignDocumentField` acting up? This *should* work but it's hard to parse these kinds of type issues until the lower level stuff is cleaned up